app/controllers/record_controller.rb

```diff
@@ -93,6 +94,20 @@ def with_session_error_handling
     @eds_error_link = rebuild_eds_full_record_link
   end
 
+  # detect and log known and unknown eds failures separately so we can
+  # better understand what is happening
+  def with_error_handling
+    yield
+  rescue EBSCO::EDS::BadRequest => e
+    if e.message.include?('Simultaneous User Limit Reached')
```

Comment:
I'm OK with this for now, but it will rapidly turn into twisty spaghetti if we discover more error conditions we want to rescue specifically. Long-term this may be happier as a hash mapping error messages to exceptions or something.

Reply:
Good idea. If it gets bigger than these cases I think it should get that suggested refactor. 👍